### PR TITLE
Pref Updates

### PR DESCRIPTION
**Pull request by rehtropsmb**
Replaced static variables tracking previous state with `pref::did_change`

### CODE REVIEW

File: src/systems/pref.cpp [hunk 1]
```diff
@@ -539,6 +539,7 @@ static constexpr char* PREF_FILENAME = "apmp";
 
 void init() {
     load_default_prefs();
+    mkb::memcpy(&s_default_pref_state, &s_prev_pref_state, sizeof(s_default_pref_state));
```
Comment (ComplexPlane) [inline src/systems/pref.cpp]: First two args should be swapped

**Review by ComplexPlane — CHANGES_REQUESTED** [review]
Looks like `fallout.cpp` has some remaining prev_pref bools too

File: src/systems/pref.cpp (hunk 1 above)
Comment (rehtropsmb) [reply src/systems/pref.cpp]: oops, fixed

**Comment by rehtropsmb** [comment]
> Looks like `fallout.cpp` has some remaining prev_pref bools too

@ComplexPlane These are all changed in my Timer PR

**Review by ComplexPlane — APPROVED** [review]

**Pull request merged**